tests/unit/resource/used_pus.cpp

```diff
@@ -0,0 +1,48 @@
+//  Copyright (c) 2017 Hartmut Kaiser
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+// Simple test verifying basic resource_partitioner functionality.
+
+#include <hpx/hpx_init.hpp>
+#include <hpx/include/resource_partitioner.hpp>
+#include <hpx/include/threads.hpp>
+#include <hpx/util/lightweight_test.hpp>
+
+#include <cstddef>
+#include <string>
+#include <utility>
+#include <vector>
+
+int hpx_main(int argc, char* argv[])
+{
+    std::size_t num_threads = hpx::resource::get_num_threads("default");
+    hpx::threads::detail::thread_pool_base& tp =
+                hpx::resource::get_thread_pool("default");
+
+    auto used_pu_mask = tp.get_used_processing_units();
+    HPX_TEST_EQ(hpx::threads::count(used_pu_mask), num_threads);
+
+    for (std::size_t t = 0; t < num_threads; ++t)
+    {
+        auto thread_mask = hpx::resource::get_partitioner().get_pu_mask(t);
+        HPX_TEST(hpx::threads::bit_or(used_pu_mask, thread_mask));
+    }
+
+    return hpx::finalize();
+}
+
+int main(int argc, char* argv[])
+{
+    std::vector<std::string> cfg = {
+        "hpx.os_threads=4"
```

Comment:
Are you sure this is a good idea? The pu mask will be 1 for each pu used. If you run this test on a single core machine, then asking for 4 threads will set the pu flag, but only once and not 4 times. So bits in pu mask will not match num threads.

Reply:
Ok, the only thing that will fail in this scenario will the test on line 25 then... I guess having at least 4 PUs available is a prerequisite of this test (this is also true for the other RP test)

Reply:
I think that's sensible. If the machine being tested on has < N cores, - just skip the test and return pass. There is no point testing unless N is useful.